metsim/metsim.py

```diff
@@ -55,87 +81,101 @@ def run(self, method, forcings):
             for i in range(len(met_data.lat)):
                 out_dict[i] = {}
                 for j in range(len(met_data.lon)):
-                    out_dict[i][j] = method.run(
-                                met_data.isel(lat=[i], lon=[j])
-                                .to_dataframe(), disagg=True)
-                    # TODO: Remove this
-                    print(i,j) 
+                    out_dict[i][j] = method.run(met_data.isel(lat=[i], lon=[j])
+                                                .to_dataframe(), MetSim.params,
+                                                disagg=True)
             # Write out
             self.write(out_dict)
 
-        # Finally, close the output
-        self.output.close()
-           
+        if MetSim.params.get('out_format', 'netcdf') == 'netcdf':
+            self.output.close()
+
 
     def find_elevation(self, lat: float, lon: float) -> float:
         """ Use the domain file to get the elevation """
         lat_idx = np.abs(self.lats - lat).argmin()
-        lon_idx = np.abs(self.lons - lon).argmin()            
+        lon_idx = np.abs(self.lons - lon).argmin()
         return self.elev[lat_idx, lon_idx]
 
-   
+
+    def update(self, new_params):
+        """Updates the global parameters dictionary"""
+        MetSim.params.update(new_params)
+
+
     def write(self, data: dict):
         """
-        Dispatch to the right function based on the configuration given 
+        Dispatch to the right function based on the configuration given
         """
         dispatch = {
                 'netcdf' : self.write_netcdf,
                 'ascii'  : self.write_ascii
                 }
-        dispatch[self.params.get('out_format', 'netcdf')](data)
-         
+        dispatch[MetSim.params.get('out_format', 'netcdf').lower()](data)
+
 
     def init_netcdf(self):
         """Initialize the output file"""
         print("Initializing netcdf...")
```

Comment:
remove debugging statements (or wrap them in some `verbose` option or logging function)